File: 128. Longest Consecutive Sequence/go/longest_consecutive_sequence.go
package main

func longestConsecutive(nums []int) int {
	d := make(map[int]bool)
	for _, i := range nums {
		d[i] = true
	}
	count := 0
	for _, n := range nums {
		_, ok := d[n-1]
		if !ok {
			t := 1
			for _, ex := d[n+t]; ex; {
				t += 1
				_, ex = d[n+t]
			}
			if t > count {
				count = t
			}
		}
	}
	return count
}

func main() {
	a := []int{100, 4, 200, 1, 3, 2}
	b := []int{0, 3, 7, 2, 5, 8, 4, 6, 0, 1}
	res := longestConsecutive(a)
	println(res)
	res = longestConsecutive(b)
	println(res)
}

File: 128. Longest Consecutive Sequence/python/longest_consecutive_sequence.py
from typing import List


class Solution:
    def longestConsecutiveS(self, nums: List[int]) -> int:
        d = set(nums)
        count = 0
        for i in nums:
            if not i-1 in d:
                t = 1
                while i+t in d:
                    t += 1
                count = max(count, t)
        return count

    def longestConsecutiveD(self, nums: List[int]) -> int:
        d = {i: True for i in nums}
        count = 0
        for i in nums:
            r = d.get(i - 1, False)
            if not r:
                t = 1
                while d.get(i + t, False):
                    t += 1
                count = max(count, t)
        return count